tools/lkl/lib/iomem.c

```diff
@@ -65,10 +68,91 @@ void *lkl_ioremap(long addr, int size)
 		return IOMEM_INDEX_TO_ADDR(index);
 
 	return NULL;
+#endif
 }
 
 int lkl_iomem_access(const volatile void *addr, void *res, int size, int write)
 {
+	/* FIXME: should be transparent with platform/ */
+#ifdef RUMPRUN
+	uint16_t mem = (unsigned long)addr;
+	int ret = 0;
+
+	if (write) {
+		if (size == 1) {
+#ifdef __x86_64__
```

Comment:
I'm guessing this means that x86-32 is not supported?

Reply:
LKL does not support x86-32, yet lkl/linux#341

Reply:
this is a messy part of lots of ifdefs and a result of what i personally tried.  this can be expanded a platform which is currently not supported (e.g., x86-32).